### PR TITLE
BT-657 Copy workflow labels to TES tags

### PR DESCRIPTION
**Pull request by breilly2**
https://broadworkbench.atlassian.net/browse/BT-657

Note that behavior is undefined if workflow labels are modified after submitting the workflow.

### CODE REVIEW

File: cromwell.example.backends/TES.conf [hunk 1]
```diff
@@ -30,4 +30,5 @@ backend {
         }
       }
     }
+  }
```
Comment (breilly2) [inline cromwell.example.backends/TES.conf:33]: Drive-by change to fix syntax.

File: supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala [hunk 2]
```diff
@@ -148,19 +148,7 @@ class TesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
...
-      description = Option(task.description),
-      inputs = Option(task.inputs),
-      outputs = Option(task.outputs),
-      resources = Option(task.resources),
-      executors = task.executors,
-      volumes = None,
-      tags = None,
-      logs = None
-    ))
+    tesTask.map(TesTask.makeTask)
```
Comment (breilly2) [inline supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala:151]: Thank you @jgainerdewar for suggesting that I reconsider how to refactor this to make writing the test easier.

File: supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesTask.scala [hunk 3]
```diff
@@ -257,6 +257,22 @@ object TesTask {
...
+      id = None,
+      state = None,
+      name = Option(tesTask.name),
+      description = Option(tesTask.description),
+      inputs = Option(tesTask.inputs),
+      outputs = Option(tesTask.outputs),
+      resources = Option(tesTask.resources),
+      executors = tesTask.executors,
+      volumes = None,
+      tags = Option(tesTask.jobDescriptor.workflowDescriptor.customLabels.asMap),
```
Comment (breilly2) [inline supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesTask.scala:272]: This is the only change to this extracted function.

File: supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesTask.scala [hunk 4]
```diff
@@ -257,6 +257,22 @@ object TesTask {
       backend_parameters = Option(backendParameters)
     )
   }
+
+  def makeTask(tesTask: TesTask): Task = {
```
Comment (breilly2) [inline supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesTask.scala:261]: I first tried to add a `fromTesTask` to a `Task` companion object, but somehow that ended up interfering with finding the implicit JSON serializer for `Task`. 🙄

File: supportedBackends/tes/src/test/scala/cromwell/backend/impl/tes/TesTaskSpec.scala [hunk 5]
```diff
@@ -47,4 +57,35 @@ class TesTaskSpec extends AnyFlatSpec with CromwellTimeoutSpec with Matchers wit
         Resources(None, None, None, Option(false), None, Option(Map.empty[String, Option[String]])
     )
   }
+
+
+  it should "copy labels to tags" in {
```
Comment (breilly2) [inline supportedBackends/tes/src/test/scala/cromwell/backend/impl/tes/TesTaskSpec.scala:62]: Still an awful lot of setup, but easier than building a suite for the TES job execution actor just to test this function.

File: supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala (hunk 2 above)
Comment (jgainerdewar) [reply supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala:151]: Nice! I'm glad the test gave us a good excuse to make this change, I like the new way much better.

**Review by jgainerdewar — APPROVED** [review]
LGTM!

**Review by kraefrei — APPROVED** [review]
